client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -86,11 +95,23 @@ export default function WPCheckoutOrderReview( {
 } ): JSX.Element {
 	const translate = useTranslate();
 	const [ isCouponFieldVisible, setCouponFieldVisible ] = useState( false );
+	const [ experiment, setExperiment ] = useState< ExperimentAssignment | null >( null );
 	const cartKey = useCartKey();
 	const { responseCart, removeCoupon, couponStatus } = useShoppingCart( cartKey );
 	const isPurchaseFree = responseCart.total_cost_integer === 0;
+	const isDesktop = useViewportMatch( 'mobile', '>=' );
 	const reduxDispatch = useDispatch();
 
+	useEffect( () => {
+		const experimentCheck = isDesktop
+			? 'registration_email_only_desktop_random_usernames'
+			: 'registration_email_only_mobile_random_usernames';
+
+		loadExperimentAssignment( experimentCheck ).then( ( experimentObject ) => {
+			setExperiment( experimentObject );
+		} );
+	}, [] );
```

Comment:
We should probably add `isDesktop` to the dependencies of `useEffect` here, or add a `// eslint-ignore-next-line react-hooks/exhaustive-deps` above it to make it explicit that we only want to run this once.